src/components/dashboard/Dashboard.js

```diff
@@ -148,6 +148,14 @@ const Dashboard = props => {
     setAvatarCenteredPosition(newAvatarCenteredPosition)
   }, [setHeaderContentWidth, setAvatarCenteredPosition])
 
+  const selectBalanceFormatter = useCallback(() => {
+    //abbreviate balance if the header shrank and if balance has more than 10 units.
+    if (!headerLarge && (balance / 100).toString().length > 10) {
+      return abbreviateValue(7)
+    }
+    return
+  }, [balance, headerLarge])
+
```

Comment:
1. no need to divide by 100. if value / 100  takes 10 units, then value takes 10 + 2 = 12 units
2. no need to convert to string. value of points of each decimal number is just it's logarithm by 10, rounded down plus 1

```suggestion
  const balanceFormatter = useMemo(() => {
    if (headerLarge || (Math.floor(Math.log10(balance)) + 1) <= 12) {
      return
    }
    
    // abbreviate balance if the header shrank and if balance has more than 10 units.
    return value => abbreviateValue(value, 7)
  }, [balance, headerLarge])

```